blog/20230219-best-8-message-queue/best-8-message-queue.md

```diff
@@ -0,0 +1,175 @@
+* kubertenes is becoming the infra of cloud native apps：Kubernetes provides a powerful ability to automatically expand and shrink applications, and dynamically adjust resources according to the load of the application, thereby achieving higher resource utilization and faster application response time.his helps enterprises save costs and improve efficiency, so more and more enterprises deploy their software on k8s.
+
+Around 2010, due to the surge of mobile Internet users, a large amount of data needs to be processed, which gave birth to the emergence of message queues such as Kafka. In 2020, due to the large number of cloud technologies adopted by enterprises and the emergence of cloud-native technologies such as kubernetes and serverless, enterprises have new needs. This time, they need a message queue with a cloud-native architecture that is truly suitable for the new infrastructure. However, message queues, which were born around 2010, are obviously powerless in the face of new infrastructure and new applications such as serverless due to different technical architectures and application scenarios. For example, kafka obviously has many problems running on kubertenes:
+
+* StatefulSet requirement: Kafka is a distributed system that requires each node to maintain state, which can make it difficult to run on Kubernetes. In particular, running Kafka on Kubernetes requires using StatefulSets, which can be more complex to manage than Deployments.
+
+* Resource consumption: Kafka requires significant resources to run, including CPU, memory, and storage. This can make it challenging to run Kafka in a scalable way on Kubernetes, where resources are typically shared among many different applications.
+
+* Networking complexity: Kafka requires a well-defined network topology in order to work properly, and this can be difficult to achieve on Kubernetes. In particular, Kafka requires that each node have a unique hostname and IP address, which can be challenging to achieve in a containerized environment.
+
+* Data locality: Kafka performs best when data is stored on the same node as the consumer that will be reading it. However, Kubernetes does not provide strong guarantees about where pods are scheduled, which can make it difficult to ensure that data is stored on the same node as the consumer.
+
+Different from virtual machines and traditional microservice architecture applications, new infrastructure such as k8s and cloud-native applications such as serverless have significantly different requirements for message queues:
+
+Fully elastic: It can make full use of the capabilities of kubernetes and automatically expand or contract as needed. Kafka can only be expanded or contracted manually, and data migration for replication is required.
```

Comment:
```suggestion
Fully elastic: It can make full use of the capabilities of Kubernetes and automatically expand or contract as needed. Kafka can only be expanded or contracted manually, and data migration for replication is required.
```